### ISSUE TITLE
"no export data for" error since v1.44

### ISSUE THREAD

**Issue by albertteoh** [golangci/golangci-lint#3376]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Running the linter returns the following error:
```
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package proj: could not load export data: no export data for "github.com/pebbe/go-proj-4/proj"
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: buildir: failed to load package proj: could not load export data: no export data for "github.com/pebbe/go-proj-4/proj"
```

After a bit of debugging, I found returning `nil`  _appears_ to prevent the error, but may just be a hack that hides a deeper problem, which I think is around understanding why github.com/pebbe/go-proj-4/proj doesn't have an exportFile. 

I'd be appreciative if someone could explain how exportFiles are generated and why it's missing for github.com/pebbe/go-proj-4/proj; or, if this is "expected", whether if my proposed change is an appropriate fix: https://github.com/golangci/golangci-lint/compare/master...albertteoh:golangci-lint:warn-instead-of-err-no-exportfile?expand=1.

Worth noting that I do not see this error when running with the docker image: `golangci/golangci-lint:v1.43-alpine`
... and this error starts appearing from `golangci/golangci-lint:v1.44-alpine`.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.50.1 built from 8926a95f on 2022-10-22T10:50:47Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
  timeout: 5m
  modules-download-mode: vendor
  go: "1.19"

issues:
  exclude-rules:
    - path: '(.+)_test\.go'
      linters:
        - gosec

linter-settings:
  depguard:
    list-type: blacklist
    include-go-root: true
    packages:
      - io/ioutil
    packages-with-error-message:
      - io/ioutil: "Use os or io instead of io/ioutil"
  goimports:
    local-prefixes: github.com/nearmap/openworld
  gosimple:
    go: "1.19"
    checks: ["all"]

linters:
  fast: false
  enable:
    - depguard
    - gofmt
    - gofumpt
    - goimports
    - gosec
    - govet
    - misspell
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19.1 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/vagrant/.cache/go-build"
GOENV="/home/vagrant/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/vagrant/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/vagrant/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19.1"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/vagrant/go/src/github.com/nearmap/openworld/go.mod"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2885735728=/tmp/go-build -gno-record-gcc-switches"```
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /vagrant/go/src/github.com/nearmap/openworld /vagrant/go/src/github.com/nearmap /vagrant/go/src/github.com /vagrant/go/src /vagrant/go /vagrant / /home/vagrant]
INFO [config_reader] Used config file .golangci.yaml
INFO [lintersdb] Active 13 linters: [depguard errcheck gofmt gofumpt goimports gosec gosimple govet ineffassign misspell staticcheck typecheck unused]
INFO [loader] Go packages loading at mode 575 (compiled_files|exports_file|files|imports|deps|name|types_sizes) took 17.15944041s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 134.341821ms
INFO [linters_context/goanalysis] analyzers took 1m15.456566948s with top 10 stages: buildir: 44.298340449s, gofumpt: 4.425482723s, goimports: 4.386180343s, gofmt: 4.236456711s, misspell: 4.019975918s, isgenerated: 3.115835806s, inspect: 1.612242537s, gosec: 1.222692737s, ctrlflow: 915.632099ms, printf: 880.704774ms
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package proj: could not load export data: no export data for "github.com/pebbe/go-proj-4/proj"
INFO [runner] processing took 2.378µs with stages: max_same_issues: 450ns, max_from_linter: 298ns, nolint: 273ns, skip_dirs: 242ns, cgo: 112ns, filename_unadjuster: 102ns, skip_files: 101ns, uniq_by_line: 99ns, path_prettifier: 99ns, autogenerated_exclude: 97ns, severity-rules: 95ns, exclude: 92ns, max_per_file_from_linter: 44ns, diff: 44ns, sort_results: 43ns, exclude-rules: 40ns, identifier_marker: 38ns, path_shortener: 38ns, source_code: 37ns, path_prefixer: 34ns
INFO [runner] linters took 19.846467904s with stages: goanalysis_metalinter: 19.846438948s
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: buildir: failed to load package proj: could not load export data: no export data for "github.com/pebbe/go-proj-4/proj"

INFO Memory: 370 samples, avg is 361.5MB, max is 1005.1MB
INFO Execution took 37.153737047s```
```

</details>

### Code example or link to a public repository

<details>

This affects a private repository. However, the dependency that seems to be the cause of this error is: github.com/pebbe/go-proj-4/proj

```go
// add your code here
```

</details>

**Comment by ldez** [comment]
Hello,

can you provide a reproducible example?

**Comment by albertteoh** [comment]
> Hello,
> 
> can you provide a reproducible example?

Thanks for the speedy response @ldez!

Sure, here's a reproducible example: https://github.com/albertteoh/repro-exportfile-missing

**Comment by ldez** [comment]
I think it's problem related to CGO

**Comment by albertteoh** [comment]
> I think it's problem related to CGO:

Yup, I think you're right.

- The reason why upgrading from v1.43 -> v1.44.0 breaks for us is because linter warnings are (correctly) surfaced as errors, I think from this PR: https://github.com/golangci/golangci-lint/pull/2471.
- I think the mistake I'm making is to run the `golangci/golangci-lint:v1.44-alpine` image against our source code (via volume mount) without successfully installing the C Proj library within the new image layer, which prevents the GC compiler from producing the export file.
- To prove this, I installed golangci-lint v1.50.1 in the "reproducer" repo (https://github.com/albertteoh/repro-exportfile-missing/commit/2862d3a73369d48f847988747594cfebf891fd31) and running `make run` succeeds without error 🥳 .

Thanks for your help @ldez!

I'm closing this issue for now and will apply my learnings to our CI pipeline.
